### PR TITLE
fix: use dragonberry fixes in cosmos-sdk

### LINKED ISSUE #6461: Dragonberry patches for agd
## What is the Problem Being Solved?

Need to fix Dragonberry for agd for Pismo launch.

## Description of the Design

Apply our ag0 change to agd.

## Security Considerations

Many.

## Test Plan

Pismo launch test plan.

### PR DESCRIPTION
**Pull request by JimLarson**
closes: #6461

## Description

Incorporate [Dragonberry](https://forum.cosmos.network/t/ibc-security-advisory-dragonberry/7702) fixes from cosmos-sdk.

### Security Considerations

See advisory above.

### Documentation Considerations

None.

### Testing Considerations

If possible, try Dragonberry repro.

### CODE REVIEW

**Comment by JimLarson** [comment]
Note: had to use `-compat=1.17` flag to `go mod tidy`.

File: go.mod [hunk 1]
```diff
@@ -130,6 +130,8 @@ replace github.com/zondax/hid => github.com/zondax/hid v0.9.1-0.20220302062450-5
 
 replace github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
 
+replace github.com/confio/ics23/go => github.com/agoric-labs/cosmos-sdk/ics23/go v0.8.0-alpha.agoric.1
```
Comment (arirubinstein) [inline go.mod:133]: was this supposed to be v0.8.8?

**Comment by arirubinstein** [comment]
Additionally, can we bump to 1.18? 

e.g. https://github.com/cosmos/gaia/blob/release/v7.0.x/go.mod#L3

File: go.mod (hunk 1 above)
Comment (JimLarson) [reply go.mod:133]: Nope - both cosmos/cosmos-sdk and agoric-labs/cosmos-sdk only use `v0.8.0`.

Comment (arirubinstein) [reply go.mod:133]: ah either will work.  was confused since you were going to tag v0.8.8 per https://github.com/agoric-labs/cosmos-sdk/pull/264

**Review by arirubinstein — APPROVED** [review]

**Pull request merged**